types/applications/baseEntitySheet.d.ts

```diff
@@ -3,7 +3,7 @@
  * See the FormApplication documentation for more complete description of this interface.
  * @typeParam O - the type of the Entity which should be managed by this form sheet
  */
-declare class BaseEntitySheet<O extends Entity = Entity> extends FormApplication<O> {
+declare class BaseEntitySheet<O extends Entity = Entity> extends FormApplication<BaseEntitySheet.Data<O>, O> {
```

Comment:
We sort of run into the same issue here if we want deriving classes of `BaseEntitySheet` to be able to specify completely different return types in their `getData`.